### ISSUE TITLE
taxonomy: allow marking observations with uncertainty such as c.f. or sp. or spp.

### ISSUE THREAD

**Issue by RmStorm** [NIVANorge/odm2-postgres-api#86]
When an observation is conducted sometimes researchers cannot identify the exact species but they can say something quite specific. For example, they can say, I observed a single species that is certainly part of the genus 'such and so'. They would denote this as `'such and so' sp.` This kind of uncertainty is distinct from the need for pseudogroups as discussed in #85.

An explanation of the usage of `c.f.`, `sp.` and `spp.` is in these tables.

Genus sp./spp. suffix | Description | Example
-- | -- | --
sp. | one (but unidentified) species of the genus | Scenedesmus sp.
spp. | two or more different species of the genus | Alexandrium spp.

cf. position | Description | Example
-- | -- | --
cf. before species name | uncertain identification of species | Gyrosigma cf. attenuatum
cf. before genus name | uncertain identification of genus | cf. Gyrosigma attenuatum

These use cases have to be accommodated in ODM2. This will require some sort of modifier on a taxon selection for an observation. Currently I'm leaning toward adding another column in the `Results` table to indicate this modifier.

This might be wrong and we should discuss in this issue if that actually suits the needs.

There is also the problem that there might be a need for something more specific then `spp.` essentially indicating that it was exactly two or three species of genus 'such and so' that were observed. The possibility to capture this in the data model should be kept in mind when making a PR for this issue.

**Comment by haakondr** [comment]
Have you thought of how to represent this in the UI?

Should we have optional checkboxes for this?

**Comment by RmStorm** [comment]
yes, UI-wise I have something like that in mind. There is many possibilities for it and we should experiment a little and do some user testing. We can create a twin issue to this one in the begroing app to focus on that. In this issue I'm mostly concerned with the representation of the information in the database.

**Comment by jeved** [comment]
I think this was spot on. I'm not too familiar with the database structure, but at least this applies to the observation, and not taxonomic registry.

And this should be picked along with the species, as a sort of modifier, in the observations form. choose between sp., spp., spp(2) spp(3) and spp(4), according to how many distinct but partially unknown taxa
